index.html

```diff
@@ -296,141 +292,162 @@
+  function createClassEntry(schedule, school, day, currentSlot, type) {
+    var scheduleObj;
+    var standardPeriods = ["0", "A", "B", "C", "D", "E", "F", "G", "H"];
+    // If it is a normal period
+    if (standardPeriods.indexOf(days[0][day][school][currentSlot]["period"]) >= 0) {
+      for (var k = 0; k < schedule["classes"].length; k++) {
+        if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
+          var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
+          var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
+          scheduleObj = { name: schedule["classes"][k]["name"],
+            teacher: teacherName,
+            teacherLastName: teacherLast,
+            room: schedule["classes"][k]["room"],
+            period: days[0][day][school][currentSlot]["period"],
+            time: ""};
+          break;
         }
-        // Add in photos if the user's schedule is being rendered
-        if (type == "full") {
-
-          // If the user is a normal user and the class is a normal class
-          if (scheduleObj.teacher != "" && schedule.grade) {
-            var sanitized_teacher_name;
-            sanitized_teacher_name = scheduleObj.teacher.toLowerCase();
-            sanitized_teacher_name = sanitized_teacher_name.replace(/ /g,"_");
-            scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
-            scheduleObj.teacherLink = "/teacher/" + sanitized_teacher_name;
+      }
+    } else { // If it is a special class, such as Lunch, Assembly, or Class Meeting
+      // Note that 0 Period Instrumental Music does not fall in this category
+      scheduleObj = {
+        name: days[0][day][school][currentSlot]["period"],
+        teacher: "",
+        room: "",
+        period: "X",
+        time: ""
+      };
+      if (scheduleObj.name == "Lunch" ||
+        scheduleObj.name == "Assembly" ||
+        scheduleObj.name == "US Community") {
 
-          // If class is a special class or free period
-          } else if (scheduleObj.period == "X" || scheduleObj.name == "Free Period") {
-            scheduleObj.teacherLink = "";
-            var image_url = scheduleObj.name.toLowerCase();
-            image_url = image_url.replace("/", "");
-            image_url = image_url.replace(/\s/g, '');
-            scheduleObj.avatar = "/images/" + image_url + ".jpg";
+        scheduleObj.room = "LPC Commons";
+      }
+    }
+    if (scheduleObj.teacher == "N/A") {
+      scheduleObj.teacher = "";
+    }
+    if (scheduleObj.room == "N/A") {
+      scheduleObj.room = "";
+    }
+    // Add in photos if the user's schedule is being rendered
+    if (type == "full") {
+      // Add the left-hand images to the schedule
+      addScheduleImages(scheduleObj, schedule);
+    } else if (type == "lite") {
+      if (letterToNum(scheduleObj.period) <= 8) { // If it's a class period
```

Comment:
This should probably look at |standardPeriods|, or that code should be switched to use letterToNum(). Seems like you are doing the same thing 2 different ways.